Misc fixes for ToC dropdown

bazzadp opened the pull request:
Misc fixes from ToC drop don added in #1393 for #985

I was running the pages through an HTML validator for another issue and it was complaining about some invalid HTML as we are using an `<a>` element inside of a `<button>` which is not allowed. I've pulled it out as a separate element and use JS to show/hide whichever as appropriate. Wish the HTML linter found these :-(

I've also hidden the dropdown arrows until JS kicks in, and done same for mobile menu.

I also fixed some of the issues discussed in https://github.com/HTTPArchive/almanac.httparchive.org/issues/985#issuecomment-727235918:
- Open ToC menu changes from tick to up arrow
- Current chapter should not be clickable
- Highlight current page with tick and yellow colouring
- Opacity change as was too light.

rviscomi commented:
The ebook part is extremely large. Do we need all that text? I'll file a new issue to discuss linking to /ebook vs the PDF.

![image](https://user-images.githubusercontent.com/1120896/99157719-83eaff80-2699-11eb-929f-ca95a805778a.png)

rviscomi commented on src/static/css/2019.css:
```diff
@@ -1304,12 +1310,28 @@ p.copyright a {
+  right: 23px;
+  -webkit-transform: rotate(-45deg);
+  -ms-transform: rotate(-45deg);
+  transform: rotate(-45deg);
```
BTW do we support `prefers-reduced-motion` yet? If/When we do, it's worth disabling the transition on this property.

bazzadp replied:
We do in JavaScript:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/e68274c79677de4a298abc0c6e3dae859ed16f52/src/static/js/chapter.js#L280-L287

Added a CSS override in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1512/commits/65eca17ee0ef5f66198886baf6c2684a8da4dce5 as based on [this web.dev article](https://web.dev/prefers-reduced-motion/#(bonus)-forcing-reduced-motion-on-all-websites).

rviscomi commented on src/templates/base/2019/base.html:
```diff
@@ -439,11 +465,11 @@
-          var parentEl = element.parentElement;
-          parentEl.removeChild(element);
-          parentEl.innerText = text;
+        document.querySelectorAll('.js-hide').forEach(element => {
```
Is this for the 0.00001% of users with JS disabled? Can this be solved with `<noscript>`?

bazzadp replied:
It's for those where JavaScript doesn't execute properly - either because they don't have JS enabled, or there's a syntax error, or unsupported JS syntax, or whatever. `<no script>` doesn't cover all of those. Can't remember who said it, but every browser is a no JS browser until it's downloaded parsed and executed.

bazzadp commented:
I've shorted the ebook considerably in https://github.com/HTTPArchive/almanac.httparchive.org/pull/1512/commits/442ad003352e994c6d0a5a338fadeafcc83a36ca

![Ebook link](https://user-images.githubusercontent.com/10931297/99158319-53bb5500-26c9-11eb-95b0-6a35fda86af6.png)

Technically we should have the Prince wording, but think the fact we have it on the ToC page and the watermark is surely enough.

rviscomi commented on src/templates/en/2019/base.html:
```diff
@@ -140,6 +140,7 @@
 {% block appendices %}Appendices{% endblock %}
 
 {% block ebook_title %}Ebook{% endblock %}
+{% block ebook_download_short %}Ebook PDF ({{ ebook_size_in_mb }}MB){% endblock %}
```
Nit: Should we try to use this pattern more to avoid retranslations for the same phrases?
```suggestion
{% block ebook_download_short %}{{ ebook_title }} PDF ({{ ebook_size_in_mb }}MB){% endblock %}
```

bazzadp replied:
That actually becomes:

```
{% block ebook_download_short %}{{ self.ebook_title() }} PDF ({{ ebook_size_in_mb }}MB){% endblock %}
```

I'm torn on this myself. Yes it's reuse but a bit more confusing for translators to read and ensure they understand and translate appropriately (e.g. change ordering to "PDF Ebook" for some languages).

Think clearer to leave as is to be honest.

rviscomi commented on src/static/css/2019.css:
```diff
@@ -972,6 +972,11 @@ p.copyright a {
   display: none;
 }
 
+div.js-show,
+#menu.js-show {
```
Are the element/ID important for specificity or can we use the plain `.js-show` selector? I'd also suggest combining with the `.hidden` rule above:

```css
.hidden,
.js-show {
  display: none;
}

bazzadp replied:
They were needed for specificity - but changed the code a bit since I wrote this and no longer needed. Hurray!

Good call in merging too.

Fixed.

MSakamaki commented on src/templates/base/2019/base.html:
```diff
@@ -278,7 +305,7 @@
           </ul>
         </nav>
-        <nav aria-label="{{ self.menu_title() }}" id="menu" aria-labelledby="menu-btn">
+        <nav aria-label="{{ self.menu_title() }}" id="menu" aria-labelledby="menu-btn" class="js-show">
```
@bazzadp 
I prefer to go to the footer navigation item rather than hide the hamburger menu if javascript is disabled.

Options:
However, it can be annoying to return to the top after moving to the footer.
For mobile, do you think it would be nice to have a link in the footer that takes you to the top of the page as well?

bazzadp replied:
Good idea - done.

I'll leave the scroll to top link for now as a bit out of scope for this and as Rick says the non-JS version should be used by very few users! I know iOS has easy scroll to top by tapping on menu in browser. Not sure if Android does but from a quick Google it appears not!

MSakamaki reviewed (commented):
Thanks for addressing the rest of the issues, @bazzadp !
I play, and comment on one thing that caught my attention.

bazzadp commented:
I also changed the ToC option to look like the button even when a link (consistent with link buttons on home page) and to continue to still show the Year and Lang menus but disabled. This should avoid any layout shift in case there is any delay with the JS kicking in.

@rviscomi to your earlier point about the `.hidden` class I got annoyed fighting specificity again and changed this to a `!important`. Don’t like to use them much but feel this is one case where it’s warranted and useful.

Think we’re good to merge now but will have another look in a bit and then merge and release.

MSakamaki reviewed (approved):
👍
